### DISCUSSION TITLE
how to use sycl_cuda_pi usm manage memory?

### DISCUSSION THREAD

**Discussion by wangzy0327** [intel/llvm#7281] [Q&A] [answered]
**Describe the bug**
how to use sycl_cuda_pi usm manage memory?

**To Reproduce**
Please describe the steps to reproduce the behavior:

are there some example for new user to study sycl usm ?

**Environment (please complete the following information):**

- OS: Ubuntu18.04
- Target device and vendor: Nvidia GPU
- DPC++ version: [e.g. commit hash or output of `clang++ --version`]
- Dependencies version: [e.g. low-level runtime versions (like NEO 20.04)]

**Additional context**

**Comment by JackAKirk** [comment, marked as answer]
cuda Managed Memory maps to USM shared memory in the cuda sycl backend as described in the SYCL 2020 spec https://github.com/intel/llvm-test-suite/blob/intel/SYCL/USM/memadvise_cuda.cpp.

For basic usage see for example this test, https://github.com/intel/llvm-test-suite/blob/intel/SYCL/USM/depends_on.cpp, which allocates shared memory, operates on it on device, then checks results on host.

You can also use memory advice hints on shared usm in the cuda backend as in this test: https://github.com/intel/llvm-test-suite/blob/intel/SYCL/USM/memadvise_cuda.cpp . Choosing good memory advice can improve performance is some cases as discussed e.g. here: https://developer.nvidia.com/blog/maximizing-unified-memory-performance-cuda/

**Comment by wangzy0327** [comment]
OK，Thank you
